Creato oggetto Match

kzmdti opened the pull request:
Aggiunte activity per la creazione della partita, mappa, e oggetto Match per la partita

kzmdti committed:
Creato oggetto Match
Aggiunte activity per la creazione della partita

kzmdti committed:
Creato oggetto Match
Aggiunte activity per la creazione della partita

Bug in MainActivity fixed

lucagiordano-visia commented on app/src/main/java/com/es/findsoccerplayers/MainActivity.java:
```diff
@@ -46,6 +66,19 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 Utils.showUnimplementedToast(this);
                 //TODO: creare entity delle impostazioni
                 return true;
+            case R.id.create_match:
```
Seguendo il layout delle activity che ho caricato sta mattina, per creare una partita dovremmo fare un FTB

lucagiordano-visia commented on app/src/main/java/com/es/findsoccerplayers/MainActivity.java:
```diff
@@ -46,6 +66,19 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 //TODO: creare entity delle impostazioni
                 return true;
+            case R.id.create_match:
+                Intent intent = new Intent(this, MatchActivity.class);
+                //If we have a last know location of the user, send it to the MatchActivity
+                if(lastLatitude > 0 && lastLongitude > 0){
+                    intent.putExtra(LATITUDE, lastLatitude);
+                    intent.putExtra(LOGITUDE, lastLongitude);
+                    startActivity(intent);
+                    // else just start the activity
```
Quando starti un nuova activity, metti i flag nell'intent per dirgli di usare l'activity già presente nello stack, senza doverla creare di nuovo

lucagiordano-visia commented on app/src/main/res/menu/map_options.xml:
```diff
@@ -0,0 +1,20 @@
+        android:title=""
+        android:icon="@android:drawable/ic_menu_add"
+        app:showAsAction="always"/>
+    <item android:id="@+id/normal_map"
+        android:title="@string/normal_map"
+        app:showAsAction="never"/>
+    <item android:id="@+id/hybrid_map"
+        android:title="@string/hybrid_map"
+        app:showAsAction="never"/>
+    <item android:id="@+id/satellite_map"
+        android:title="@string/satellite_map"
+        app:showAsAction="never"/>
+    <item android:id="@+id/terrain_map"
+        android:title="@string/terrain_map"
+        app:showAsAction="never"/>
```
io terrei solo normal view e satelite, poichè normal ha tutti i segnalini delle posizioni che servono a noi, mentre satelite può tornare utile quando non sai dov'è il campo e vuoi cercarlo dal satellite. Le altre due non servono a molto a mio parere

lucagiordano-visia commented on app/src/main/res/menu/map_options.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item android:id="@+id/confirm_position"
+        android:title=""
+        android:icon="@android:drawable/ic_menu_add"
+        app:showAsAction="always"/>
```
Per la conferma della posizione penso sia più carino un FTB con una spunta, se si può fare

lucagiordano-visia commented on app/src/main/res/layout/activity_match.xml:
```diff
@@ -0,0 +1,140 @@
+            app:layout_constraintStart_toEndOf="@+id/imageViewTime"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <TextView
+            android:id="@+id/date_text"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="50dp"
+            android:layout_marginTop="16dp"
+            android:textSize="15dp"
+            android:text="TextView"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/date_hour" />
+
+        <TextView
+            android:id="@+id/time_text"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="16dp"
+            android:layout_marginEnd="24dp"
+            android:text="TextView"
+            android:textSize="15dp"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/date_hour" />
```
Io qui troverei più intuitivo mettere due stringhe "SELEZIONA DATA" "SELEZIONA ORA" al posto di mettere data e ora attuali, un po' come ho fatto su Registe

lucagiordano-visia commented on app/src/main/res/layout/activity_match.xml:
```diff
@@ -0,0 +1,140 @@
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MatchActivity">
+
```
ci sarebbe da aggiungere la toolbat con la freccia di back come nelle altre activity

lucagiordano-visia commented on app/src/main/res/layout/activity_maps.xml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
```
ci sarebbe da aggiungere la freccia di back come nelle altre activity

kzmdti committed:
Creato oggetto Match
Aggiunte activity per la creazione della partita

Bug in MainActivity fixed

Aggiunto savedInstanceState in MatchActivity
Agginto FAB in MainActivity e MatchActivity
Diversi Bug fixed

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/MainActivity.java:
```diff
@@ -41,10 +44,28 @@ protected void onCreate(Bundle savedInstanceState) {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(MainActivity.this, MatchActivity.class);
+                //If we have a last know location of the user, send it to the MatchActivity
```
"known"

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/MainActivity.java:
```diff
@@ -41,10 +44,28 @@ protected void onCreate(Bundle savedInstanceState) {
+                    // else just start the activity
+                } else {
+                    startActivity(intent);
+                }
```
togli l'`else` e scrivi dentro l'`if` solo i `putExtra`

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/MyPickers/DatePickerFragment.java:
```diff
@@ -33,7 +36,7 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
     // setTheDate di MatchActivity
     public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
         setTheDate(dayOfMonth, month+ 1, year);
```
lo chiamerei `setDate`

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/MatchActivity.java:
```diff
@@ -28,47 +31,70 @@
-    public final String PLACE_NAME = "place name";
-    public final int MATCH_A_REQUEST_CODE = 42;
+
+    private final String LOGITUDE = "longitude";
```
"LONGITUDE"

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/MainActivity.java:
```diff
@@ -65,4 +98,46 @@ public void onBackPressed() {
+                        lastLongitude = mlastLocation.getLongitude();
+                    } else {
+                        // To delete in definitive build
+                        Toast.makeText(MainActivity.this, R.string.gps_disabled_toast, Toast.LENGTH_SHORT).show();
```
TODO

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/MyPickers/NumberPickerDialog.java:
```diff
@@ -0,0 +1,40 @@
+
+
+        AlertDialog.Builder alertDialog = new AlertDialog.Builder(getActivity());
+        alertDialog.setTitle("Quanti giocatori ti mancano?").setView(np).setPositiveButton("OK", new DialogInterface.OnClickListener() {
```
hardcoded string

d-u-d-e commented on app/src/main/res/values/strings.xml:
```diff
@@ -39,4 +39,24 @@
+    <string name="gecodo_failed">Position name not found. Geocode didn\'t work.</string>
+    <string name="default_position_name">My Position</string>
+
+
```
diamogli un nome coerente con gli altri, aggiungendo davanti il nome dell'activity come negli altri casi

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/MapsActivity.java:
```diff
@@ -0,0 +1,237 @@
+                    }
+                } catch (Exception e){
+                    e.printStackTrace();
+                    Toast.makeText(MapsActivity.this, R.string.gecodo_failed, Toast.LENGTH_SHORT).show();
```
se fallisce assegna a `placename` semplicemente la posizione di default, senza toast o print dello stack trace
